### ISSUE TITLE
MT5 fallback with mt5_contrib=True under xp=jax (follow-up to #196)

### ISSUE THREAD

**Issue by gschnabel** [IAEA-NDS/endf-userpy#215]
Follow-up to #196 / PR #214. Roadmap: #198.

## Current state

PR #214 short-circuited the MT5 fallback branch of \`get_reaction_xs\` for files without MF6/MT=5 (typical: Be-9 (n,g), most light nuclei), so \`mt5_contrib=True\` no longer crashes under \`xp=jax\` on those files.

For files that DO carry MF6/MT=5 (heavier evaluations), \`mt5_contrib=True\` under \`xp=jax\` now raises \`NotImplementedError\` with a suggestive message. Users must pass \`mt5_contrib=False\` to skip the fallback.

### Approach A -- xp-native port (preserves grad through MT5)

Rewrite the mask logic tracer-safely:

\`\`\`python
mt5_xs_all = compute_xs_mt5_contrib(endf_dict, mt, energies_in, xp=xp)
xs = xp.where(cur_xs == 0.0, xs + mt5_xs_all, xs)
\`\`\`

Compute the MT5 contribution on all Es rather than the concrete \`eincs_sel\` slice, then let \`xp.where\` do the "only where direct is zero" masking.

Requires porting \`compute_xs_mt5_contrib\` (and its downstream \`compute_yields\` for MT=5 -- \`mf3_interp.compute_cross_section\` is already xp-agnostic per PR #193) to accept \`xp\`.

Cost: one extra MT=5 evaluation per query (small).

Benefit: \`jax.grad\` propagates through the MT5 backfill.

### Approach B -- \`jax.pure_callback\` wrap (no grad through MT5)

Same pattern as MF6 LAW=1 pcb dispatch (#194). Run the numpy MT5 fallback inside a \`jax.pure_callback\`, treat the returned contribution as a black-box additive value.

Cost: much smaller than A; just wrap the existing numpy branch.

Benefit: \`jax.grad\` still works through the direct MF3 path. Downside: MT5 backfill contribution has zero gradient wrt E, which is wrong if MT5 is the dominant contribution physically.

## When to prioritise

- Approach B is enough for the common case where MT5 catches only sparse contributions above the specific-MT threshold cluster (light nuclei, medium-heavy nuclei outside the actinide domain). Gradient is dominated by the direct MF3 path.
- Approach A is required for the actinide (n,3n)/(n,4n) or high-E residual production channels where MT5 can be the dominant contribution.

Recommendation: defer both until a real user hits the limitation and can tell us which reaction / file combination they need, so we know which approach applies. In the meantime, \`mt5_contrib=False\` is a working escape hatch.

## Test coverage today

\`tests/test_mf3_xs_autodiff_wrt_E.py::test_grad_wrt_E_ngamma_mt5_default_works_on_file_without_mt5\` pins the no-MT=5 short-circuit. When Approach A/B lands, add:

- Positive assertion on a file WITH MT=5 that grad wrt E is finite under \`mt5_contrib=True\` + \`xp=jax\`.
- FD-match on a reaction where MT5 contributes substantially.

## Related

- #196 (parent): MT5 fallback not xp-agnostic (partially fixed by #214).
- #194: MF6 LAW=1 pcb dispatch (template for Approach B).
- Roadmap: #198.
